src/dataset.py

```diff
@@ -5,66 +5,101 @@
-      cropped_img_path = Path(self.base_dir, cropped_img_path)
-      cropped_label_path = Path(self.base_dir, cropped_label_path)
-
-      cropped_img = np.load(cropped_img_path)
-      mask_path = cropped_label_path if (self.mode == 'train') else mask_path
-      image_path = cropped_img_path
-
-    img = np.load(image_path)
-    if self.img_transform is not None:
-      img = self.img_transform(img)
-    else:
-      img = T.ToTensor()(img)
-
-    if (self.borders) and (not pd.isnull(border_path)):
-      border_path = Path(self.base_dir, border_path)
-      border = np.load(border_path)
-      border = np.expand_dims(border, axis=0)
-      img = np.concatenate((img, border), axis=0)
-
-    return img, np.load(mask_path).astype(np.float32)
-
-  def __len__(self):
-    return len(self.data)
-
-
-def loader(data_opts, train_opts, mode="train"):
+    def __init__(self, base_dir, data_file, channels_to_inc=None, img_transform=None,
+                 mode='train', borders=False, use_cropped=True, use_snow_i=False,
+                 mask_used='glacier'):
+        super().__init__()
+        self.base_dir = base_dir
+        data_path = Path(base_dir, data_file)
+        self.data = pd.read_csv(data_path)
+        self.data = self.data[self.data.train == mode]
+        if mask_used == 'debris_glaciers':
+            self.data = self.data[self.data.debris_perc > 0]
+        self.img_transform = img_transform
+        self.borders = borders
+        self.use_cropped = use_cropped
+        self.use_snow_i = use_snow_i
+        self.channels_to_inc = channels_to_inc
+        self.mode = mode
+        self.mask_used = mask_used
+
+    def __getitem__(self, i):
+        pathes = ['img_path', 'mask_path', 'border_path']
+        image_path, mask_path, border_path = self.data.iloc[i][pathes]
+
+        image_path = Path(self.base_dir, image_path)
+        mask_path = Path(self.base_dir, mask_path)
+
+        if self.use_cropped:
+            cropped_pathes = ['cropped_path', 'cropped_label']
+            cropped_img_path, cropped_label_path = self.data.iloc[i][cropped_pathes]
+            cropped_img_path = Path(self.base_dir, cropped_img_path)
+            cropped_label_path = Path(self.base_dir, cropped_label_path)
+
+            cropped_img = np.load(cropped_img_path)
+            mask_path = cropped_label_path if (
+                self.mode == 'train') else mask_path
+            image_path = cropped_img_path
+
+        img = np.load(image_path)
+        img = T.ToTensor()(img)
+
+        if self.channels_to_inc is not None:
+            img = img[self.channels_to_inc]
+
+        if (self.borders) and (not pd.isnull(border_path)):
+            border_path = Path(self.base_dir, border_path)
+            border = np.load(border_path)
+            border = np.expand_dims(border, axis=0)
+            img = np.concatenate((img, border), axis=0)
+            img = torch.from_numpy(img)
+
+
```

Comment:
Can you remove some of the unnecessary whitespace here and in the next few lines?